src/components/Preview.js

```diff
@@ -24,34 +34,50 @@ class SnapguidistPreview extends Component {
     }
   }
 
-  runTest(update) {
-    this.setState({ isFetching: true })
-
-    api
-      .runTest(this.context.name, this.example, update)
-      .then(response => this.setState({ response, isFetching: false }))
+  componentWillUnmount() {
+    this.forgetSnapguidist()
   }
 
   evalInContext(code) {
     const result = this.props.evalInContext(code)
 
     const wrapper = (state, setState, callback) => {
       this.example = result(state, setState, callback)
-
       return this.example
     }
 
     return wrapper
   }
 
+  handleSnapshotResponse({ name, response } = {}) {
+    if (this.context.name === name) {
+      this.setState({ response, isFetching: false })
+    }
+  }
+
+  runTest(update) {
+    const {
+      name,
+      snapguidist: { runTest } = {},
+    } = this.context
+
+    const { isFetching, response } = runTest(name, this.example, update)
```

Comment:
1. I'm not sure how to change that, also in the evolution of the feature this part changes a little. If you agree I'll wait for the other WIP features to re-discuss that point.
2. I liked it! I'll refactor it.
3. Totally agree, I'll change the code